tests/helpers.py

```diff
@@ -28,3 +31,63 @@ def fixture(path=None):
         return FIXTURE_PATH / path
     else:
         return FIXTURE_PATH
+
+
+def assert_deepequals(a, b, ignore_paths=None, _path=None):
+    """Compare objects a and b keeping track of object path for error reporting.
+
+    Keyword arguments:
+    a -- Object a
+    b -- Object b
+    ignore_paths -- List of object paths (delimited by .)
+
+    Example:
+    assert_deepequals({
+        "poetry-version": "1.0.0a3",
+        "content-hash": "example",
+    }, {
+      "metadata": {
+        "poetry-version": "1.0.0a4",
+        "content-hash": "example",
+      }
+    }, ignore_paths=set(["metadata.poetry-version"]))
+    """
+
+    _path = _path if _path else tuple()
+    ignore_paths = ignore_paths if ignore_paths else set()
+    path = ".".join(_path)
+    err = ValueError("{path}: {a} != {b}".format(path=path, a=a, b=b))
+
+    def make_path(entry):
+        return _path + (str(entry),)
+
+    if isinstance(a, list):
+        if not isinstance(b, list) or len(a) != len(b):
+            raise err
+
+        for idx, vals in enumerate(zip(a, b)):
+            p = make_path(idx)
+            if ".".join(p) not in ignore_paths:
+                assert_deepequals(*vals, _path=p, ignore_paths=ignore_paths)
+
+    elif isinstance(a, dict):
+        if not isinstance(b, dict):
+            raise err
+
+        for key in set(list(a.keys()) + list(b.keys())):
```

Comment:
`keys()` views are already set-like, so you can reduce this to
```
for key in a.keys() + b.keys():
```
:)

Reply:
No, you cannot:
``` python
>>> d = {}
>>> d.keys() + d.keys()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'dict_keys' and 'dict_keys'
>>>
```